### PR TITLE
Fix docstrings in nullable.jl

### PR DESCRIPTION
**Pull request by xorJane**
This moves inline and edits the docstring for `Nullable`, adding doctests and changing the function signature. Also fixes the doctests for `isnull`. Thank you in advance for reviewing :)

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks! Looks like I indeed missed these when changing the meaning of the second argument to `Nullable`.

File: base/nullable.jl [hunk 1]
```diff
@@ -3,6 +3,27 @@
 immutable NullException <: Exception
 end
 
+"""
+    Nullable(x, hasvalue::Bool=true)
+
+When `hasvalue` is true, wrap value `x` in an object of type `Nullable`, which indicates
```
Comment (xorJane) [inline base/nullable.jl]: >I think I would phrase it slightly differently: as the original docstring said (even though it was incomplete), x is always wrapped in the resulting Nullable. The hasvalue argument will determine whether the value will be considered present or not, but it's actually always there in the value field of the result (cf. #16923).

I appreciate the explanation and reference. I understand the behavior better now.

>The hasvalue argument is useful only in quite specific cases, so I'd mention it only in a separate paragraph, to keep the explanation for the one-argument form simple. The fist paragraph could almost be kept as it is currently.

Now the `hasvalue` argument is addressed in the second paragraph of the docstring only. I've included an additional doctest to better illustrate behavior when `hasvalue` is `false`.

Thanks!

File: base/nullable.jl [hunk 2]
```diff
@@ -3,6 +3,34 @@
...
+`Nullable(x)` yields a non-empty wrapper and `Nullable{T}()` yields an empty instance of a
+wrapper that might contain a value of type `T`.
+
+`Nullable(x, false)` yields `Nullable{typeof(x)}()` with `x` stored in the result's `value`
+field.
+
+# Examples
+
+```jldoctest
+julia> x = 1; Nullable(x)
```
Comment (nalimilan) [inline base/nullable.jl]: Why not replace `x` with `1` directly in the expressions?

File: base/nullable.jl [hunk 3]
```diff
@@ -3,6 +3,34 @@
...
+Nullable{Int64}(1)
+
+julia> Nullable{Int64}()
+Nullable{Int64}()
+
+julia> Nullable(x, false)
+Nullable{Int64}()
+
+julia> dump(Nullable(x, false))
```
Comment (nalimilan) [inline base/nullable.jl]: Good idea to show this indeed!

File: base/nullable.jl (hunk 2 above)
Comment (xorJane) [inline base/nullable.jl]: Good point. Fixed it!

**Review by nalimilan — APPROVED** [review]
Thanks!

**Pull request merged**